modules/apps/27-interchain-accounts/controller/keeper/keeper.go

```diff
@@ -146,6 +147,21 @@ func (k Keeper) GetOpenActiveChannel(ctx sdk.Context, connectionID, portID strin
 	return "", false
 }
 
+// IsActiveChannelClosed retrieves the active channel from the store and returns true if the channel state is CLOSED, otherwise false
+func (k Keeper) IsActiveChannelClosed(ctx sdk.Context, connectionID, portID string) bool {
+	channelID, found := k.GetActiveChannelID(ctx, connectionID, portID)
+	if !found {
+		return false
+	}
+
+	channel, found := k.channelKeeper.GetChannel(ctx, portID, channelID)
+	if found && channel.State == channeltypes.CLOSED {
+		return true
+	}
+
+	return false
```

Comment:
Don't feel super strongly about this but it can be simplified

```suggestion
	return found && channel.State == channeltypes.CLOSED
```

Reply:
💯